Add documentation for Firebolt adapter

ericf-firebolt opened the pull request:
## Description & motivation
<!---
Added an entry for the Firebolt adapter, dbt-firebolt, to available-adapters. Also added firebolt-profile.md and firebolt-config.md files for dbt-firebolt.
-->

## Pre-release docs
Is this change related to an unreleased version of dbt?
- [ ] Yes: please
    - update the base branch to `next`
    - add Changelog components: `<Changelog>[New/Changed] in v0.x.0</Changelog>`
    - add links to the "New and changed documentation" section of the latest [Migration Guide](../website/docs/docs/guides/migration-guide)
- [x] No: please ensure the base branch is `current`
- [ ] Unsure: we'll let you know!

## Checklist
If you added new pages (delete if not applicable):
- [x] The page has been added to `website/sidebars.js`
- [x] The new page has a unique filename

ericf-firebolt committed:
Added Firebolt to available-adapters.md.

ericf-firebolt committed:
Added Firebolt configs.

ericf-firebolt committed:
Add Firebolt adapter profile.

ericf-firebolt committed:
Added firebolt profile and config files to sidebars.js.

swanderz commented:
cc: @kevinmarr

dataders commented on website/docs/docs/available-adapters.md:
```diff
@@ -43,6 +43,7 @@ These adapter plugins are contributed and maintained by members of the community
 | Oracle Database        | [Profile Setup](oracle-profile)       | Oracle 11+                | `pip install dbt-oracle`     |
 | Dremio                 | [Profile Setup](dremio-profile)       | Dremio 4.7+               | `pip install dbt-dremio`     |
 | ClickHouse             | [Profile Setup](clickhouse-profile)   | ClickHouse 20.11+         | `pip install dbt-clickhouse` |
+| Firebolt               | [Profile Setup](firebolt-profile), [Configuration](firebolt-configs) | Firebolt | `pip install dbt-firebolt`   |
```
```suggestion
```

dataders commented on website/docs/docs/available-adapters.md:
```diff
@@ -29,7 +29,7 @@ These adapter plugins are built and maintained by the same people who build and
 | Materialize ([dbt-materialize](https://github.com/MaterializeInc/materialize/blob/main/misc/dbt-materialize))  | [Profile Setup](materialize-profile) | `pip install dbt-materialize` |
 | Rockset ([dbt-rockset](https://github.com/rockset/dbt-rockset))  | [Profile Setup](rockset-profile) | `pip install dbt-rockset` |
-| Starburst & Trino ([dbt-trino](https://github.com/starburstdata/dbt-trino)) | [Profile Setup](trino-profile)  | `pip install dbt-trino` 
+| Starburst & Trino ([dbt-trino](https://github.com/starburstdata/dbt-trino)) | [Profile Setup](trino-profile)  | `pip install dbt-trino`
```
```suggestion
| Starburst & Trino ([dbt-trino](https://github.com/starburstdata/dbt-trino)) | [Profile Setup](trino-profile)  | `pip install dbt-trino` |
| Firebolt               | [Profile Setup](firebolt-profile), [Configuration](firebolt-configs) | Firebolt | `pip install dbt-firebolt`   |
```

dataders commented on website/docs/reference/warehouse-profiles/firebolt-profile.md:
```diff
@@ -0,0 +1,86 @@
+
+:::
+
+## Overview of dbt-synapse
```
```suggestion
## Overview of dbt-firebolt
```

dataders commented on website/docs/reference/warehouse-profiles/firebolt-profile.md:
```diff
@@ -0,0 +1,86 @@
+**Source:** [Github](https://github.com/firebolt-db/dbt-firebolt)  
+**dbt Cloud:** Not Supported  
+
+![dbt-synapse stars](https://img.shields.io/github/stars/firebolt-db/dbt-firebolt?style=for-the-badge)
```
```suggestion
![dbt-firebolt stars](https://img.shields.io/github/stars/firebolt-db/dbt-firebolt?style=for-the-badge)
```

dataders reviewed (changes requested):
small tweaks:
1. dbt-firebolt is vendor supported not a community adapter
2. correcting a few mentions of synapse

ericf-firebolt committed:
Apply suggestions from code review

Typos and moving Firebolt adapter to vendor-supported from community-supported.

Co-authored-by: Anders <swanson.anders@gmail.com>

kevinmarr commented on website/docs/docs/available-adapters.md:
```diff
@@ -29,7 +29,8 @@ These adapter plugins are built and maintained by the same people who build and
 | Rockset ([dbt-rockset](https://github.com/rockset/dbt-rockset))  | [Profile Setup](rockset-profile) | `pip install dbt-rockset` |
-| Starburst & Trino ([dbt-trino](https://github.com/starburstdata/dbt-trino)) | [Profile Setup](trino-profile)  | `pip install dbt-trino` 
+| Starburst & Trino ([dbt-trino](https://github.com/starburstdata/dbt-trino)) | [Profile Setup](trino-profile)  | `pip install dbt-trino` |
+| Firebolt               | [Profile Setup](firebolt-profile), [Configuration](firebolt-configs) | Firebolt | `pip install dbt-firebolt`   |
```
This row has too many columns and is not rendering properly

dataders replied:
good catch, i think this is the fix
```suggestion
| Firebolt               | [Profile Setup](firebolt-profile), [Configuration](firebolt-configs) | `pip install dbt-firebolt`   |
```

kevinmarr replied:
Nice. Can we link to the repo like the others do?

kevinmarr commented on website/docs/reference/resource-configs/firebolt-configs.md:
```diff
@@ -0,0 +1,279 @@
+id: "firebolt-configs"
+---
+
+### Setup Recommendations
```
I don't think this information belongs on the configs doc. The configs doc should be purely model configs. I'd remove this section altogether and just leave it in the adapter repo's README.

ericf-firebolt replied:
This _is_ a config file, and these _are_ configuration suggestions. At the same time, it seems kind of granular. @swanderz and I had a discussion  yesterday about what to include here vs. the readme. I think you two should discuss this before I go removing anything.

kevinmarr replied:
In dbt docs, this page will be listed as a sub-page of "Adapter-specific configs". `quote_columns` isn't adapter specific.

amychen1776 replied:
I personally think this would be good to leave in the adapter-specific config because you can give a warning as to why the warning is happening and how to fix it (especially because we don't have adapter specific troubleshooting pages)

kevinmarr replied:
Okay, that's fine with me. Let's just leave the quote_columns thing in there then.

kevinmarr commented on website/docs/reference/resource-configs/firebolt-configs.md:
```diff
@@ -0,0 +1,279 @@
+
+In her `.dbt/profiles.yml`, Sharar sets `schema=sh`, whereas Eric sets `schema=er` in his. When each runs the `customers` model, the models will land in the database as tables named `sh_customers` and `er_customers`, respectively. When running dbt in production, you would use yet another `profiles.yml` with nothing set in the `schema` field (or any string of your choice).
+
+## Feature Support
```
This information also does not belong here - let's remove.

kevinmarr commented on website/docs/reference/resource-configs/firebolt-configs.md:
```diff
@@ -0,0 +1,279 @@
+
+## Using dbt-firebolt
+
+### Model Configuration for Fact Tables
```
keep this section

kevinmarr commented on website/docs/reference/resource-configs/firebolt-configs.md:
```diff
@@ -0,0 +1,279 @@
+
+A dbt model can be created as a Firebolt fact table and configured using the following syntax:
+
+Project file syntax
```
We should mark this up such that the three code examples display as tabs in the same code snippet. See https://docs.getdbt.com/reference/model-configs for an example.

kevinmarr commented on website/docs/reference/resource-configs/firebolt-configs.md:
```diff
@@ -0,0 +1,279 @@
+
+A dbt model can be created as a Firebolt dimension table and configured using the following syntax:
+
+Project file syntax
```
See comment above about consolidating the three examples into one code block

kevinmarr commented on website/docs/reference/resource-configs/firebolt-configs.md:
````diff
@@ -0,0 +1,279 @@
+) }}
+```
+
+### How Aggregating Indexes and Join Indexes Are Named
````
Let's pull this out and leave it in the adapter repo README

kevinmarr commented on website/docs/reference/resource-configs/firebolt-configs.md:
```diff
@@ -0,0 +1,279 @@
+
+For example, a join index could be named `my_users__id__join_1633504263` and an aggregating index could be named `my_orders__order_date__aggregating_1633504263`.
+
+### Managing Ingestion via External Tables
```
Pull out and leave in README

kevinmarr commented on website/docs/reference/warehouse-profiles/firebolt-profile.md:
```diff
@@ -0,0 +1,86 @@
+
+## Overview of dbt-firebolt
+
+**Maintained by:** Community  
```
Firebolt

dataders replied:
```suggestion
**Maintained by:** Firebolt  
```

kevinmarr commented on website/docs/reference/warehouse-profiles/firebolt-profile.md:
```diff
@@ -0,0 +1,86 @@
+
+</File>
+
+### Setup Recommendations
```
remove section

ericf-firebolt committed:
Remove extraneous column

Co-authored-by: Anders <swanson.anders@gmail.com>

ericf-firebolt commented on website/docs/docs/available-adapters.md:
```diff
@@ -29,7 +29,8 @@ These adapter plugins are built and maintained by the same people who build and
 | Rockset ([dbt-rockset](https://github.com/rockset/dbt-rockset))  | [Profile Setup](rockset-profile) | `pip install dbt-rockset` |
-| Starburst & Trino ([dbt-trino](https://github.com/starburstdata/dbt-trino)) | [Profile Setup](trino-profile)  | `pip install dbt-trino` 
+| Starburst & Trino ([dbt-trino](https://github.com/starburstdata/dbt-trino)) | [Profile Setup](trino-profile)  | `pip install dbt-trino` |
+| Firebolt               | [Profile Setup](firebolt-profile), [Configuration](firebolt-configs) | Firebolt | `pip install dbt-firebolt`   |
```
Copy/paste errors: this had been in a different section.

```suggestion
| Firebolt ([dbt-firebolt](https://github.com/firebolt-db/dbt-firebolt))  | [Profile Setup](firebolt-profile), [Configuration](firebolt-configs) | `pip install dbt-firebolt`  |
```

ericf-firebolt committed:
Add tab groups to firebolt-configs.md.

ericf-firebolt committed:
Add config tab group to firebolt-configs.md.

dataders commented on website/docs/reference/resource-configs/firebolt-configs.md:
```diff
@@ -0,0 +1,279 @@
+| Aggregating indexes          | :white_check_mark: |
+| Join indexes                 | :white_check_mark: |
+| Partitioned tables           | :x: |
+
```
```suggestion
```

dataders commented on website/docs/reference/resource-configs/firebolt-configs.md:
````diff
@@ -0,0 +1,279 @@
+          columns:
+            - name: <column-name>
+              data_type: <type>
+```
````
```suggestion
```

dataders commented on website/docs/reference/warehouse-profiles/firebolt-profile.md:
```diff
@@ -0,0 +1,86 @@
+
+For an example of how this works, let’s say Shahar and Eric are both working on the same project.
+
+In her `.dbt/profiles.yml`, Sharar sets `schema=sh`, whereas Eric sets `schema=er` in his. When each runs the `customers` model, the models will land in the database as tables named `sh_customers` and `er_customers`, respectively. When running dbt in production, you would use yet another `profiles.yml` with nothing set in the `schema` field (or any string of your choice).
```
```suggestion
```

dataders commented on website/docs/reference/resource-configs/firebolt-configs.md:
````diff
@@ -0,0 +1,279 @@
+      ...
+    ]
+) }}
+```
````
this should do it? the preview rendering is failing bc this is complicated so [here's a gist for it](https://gist.github.com/swanderz/01e1145e596801dfbd95bfba1a2c17da#file-excerpt_1-md)
```suggestion
<Tabs
  defaultValue="Project"
  values={[
    { label: 'Project file', value: 'Project'},
    { label: 'schema file', value:'Property'},
    { label: 'config block', value: 'Config'},
    ]
}>

<TabItem value="Project">
<File name='dbt_project.yml'>

```yml
models:
  <resource-path>:
    +materialized: table
    +table_type: fact
    +primary_index: [ <column-name>, ... ]
    +indexes:
      - type: aggregating
        key_column: [ <column-name>, ... ]
        aggregation: [ <agg-sql>, ... ]
      ...
```

</File>
</TabItem>

<TabItem value="Project">
<File name='schema.yml'>

```yml
models:
  - name: <model-name>
    config:
      materialized: table
      table_type: fact
      primary_index: [ <column-name>, ... ]
      indexes:
        - type: aggregating | join
          key_column: [ <column-name>, ... ]
          aggregation: [ <agg-sql>, ... ]
        ...
```

</File>
</TabItem>
<TabItem value="Config">
<File name='my_model.sql'>

```
{{ config(
    materialized = "table"
    table_type = "fact"
    primary_index = [ "<column-name>", ... ],
    indexes = [
      {
        type = "aggregating"
        key_column = [ "<column-name>", ... ],
        aggregation = [ "<agg-sql>", ... ],
      },
      ...
    ]
) }}
```

</File>

</TabItem>

</Tabs>
```

dataders commented on website/docs/reference/resource-configs/firebolt-configs.md:
````diff
@@ -0,0 +1,279 @@
+models:
+  <resource-path>:
+    +materialized: table
+    +table_type: dimension
+    +indexes:
+      - type: join
+        join_column: <column-name>
+        dimension_column: [ <column-name>, ... ]
+      ...
+```
+
+Property file syntax
+```yml
+models:
+  - name: <model-name>
+    config:
+      materialized: table
+      table_type: dimension
+      indexes:
+        - type: join
+          join_column: <column-name>
+          dimension_column: [ <column-name>, ... ]
+        ...
+```
+
+Config block syntax
+```
+{{ config(
+    materialized = "table"
+    table_type = "dimension",
+    indexes = [
+      {
+        type = "join",
+        join_column = "<column-name>",
+        dimension_column: [ "<column-name>", ... ]
+      },
+      ...
+    ]
+) }}
+```
````
[second gist](https://gist.github.com/swanderz/01e1145e596801dfbd95bfba1a2c17da#file-excerpt_2-md) 

```suggestion
<Tabs
  defaultValue="Project"
  values={[
    { label: 'Project file', value: 'Project'},
    { label: 'schema file', value:'Property'},
    { label: 'config block', value: 'Config'},
    ]
}>

<TabItem value="Project">
<File name='dbt_project.yml'>

```yml
models:
  <resource-path>:
    +materialized: table
    +table_type: dimension
    +indexes:
      - type: join
        join_column: <column-name>
        dimension_column: [ <column-name>, ... ]
      ...
```

</File>

</TabItem>

<TabItem value="Project">

<File name='schema.yml'>


```yml
models:
  - name: <model-name>
    config:
      materialized: table
      table_type: dimension
      indexes:
        - type: join
          join_column: <column-name>
          dimension_column: [ <column-name>, ... ]
        ...
```

</File>

</TabItem>

<TabItem value="Config">

<File name='my_model.sql'>

Config block syntax
```
{{ config(
    materialized = "table"
    table_type = "dimension",
    indexes = [
      {
        type = "join",
        join_column = "<column-name>",
        dimension_column: [ "<column-name>", ... ]
      },
      ...
    ]
) }}
```

</File>

</TabItem>

</Tabs>

```

dataders reviewed (changes requested):
@ericf-firebolt, @kevinmarr and I met this today to discuss what does and doesn't belong in dbt's docs. the conclusion is that the following belongs in dbt's docs:
- profile set up
- how to configure specifically models
what will live in dbt-firebot's readme:
- advice on how to set up models generally

perhaps the only thing worth adding to these docs, would be hyperlinks to the general project config that we've decided to leave out

swanderz commented:
@ericf-firebolt from the [build log](https://app.netlify.com/sites/docs-getdbt-com/deploys/61af1597d81d9100084d8af1), it there's a failure related to a missing tag in `firebolt-configs.md`? i must have botched the edit...

```
12:08:41 AM: SyntaxError: /opt/build/repo/website/docs/reference/resource-configs/firebolt-configs.md: Expected corresponding JSX closing tag for <agg-sql>. (129:0)
12:08:41 AM:   127 |       ...
12:08:41 AM:   128 | ```
> 129 | </File>
      | ^
  130 | </TabItem>
  131 | <TabItem value="property-yaml">
  132 | <File name='models/properties.yml'>
```

ericf-firebolt committed:
Fixing formatting/tag closure issues.

ericf-firebolt committed:
Added closing > to closing tag in firebolt-config.md.

ericf-firebolt committed:
Cleaning up a little more firebolt-config.md: added some commas and changed yml to yaml a couple of times, and put spaces before triple-tics.

ericf-firebolt committed:
More little tweaks: spelling out yaml after triple ticks, attempting to figure out usage of ... in rendered yaml.

ericf-firebolt committed:
Trying to track down missing closing tag.

ericf-firebolt committed:
Deleting code from end of file to track down error.

amychen1776 commented on website/docs/reference/resource-configs/firebolt-configs.md:
```diff
@@ -0,0 +1,285 @@
+
+## Feature Support
+
+The table below shows which dbt and Firebolt features are supported by the adapter. dbt-firebolt is under active development and will be gradually unlocking more features over time.
```
This section should be under the profile page. [See here Materialize for reference](https://docs.getdbt.com/reference/warehouse-profiles/materialize-profile)

kevinmarr replied:
I disagree; I think it is unexpected to find this information in a doc entitled "Firebolt Profile". The expectation is to find information only pertaining to creating profiles for Firebolt connections. My preference is to have it in the repo readme for dbt-firebolt and not in dbt docs at all.

amychen1776 commented on website/docs/reference/warehouse-profiles/firebolt-profile.md:
```diff
@@ -0,0 +1,86 @@
+---
+
+
+:::info Community plugin
```
This should be removed since this is a vendor supported adapter.

dataders replied:
```suggestion
```

amychen1776 commented on website/docs/reference/warehouse-profiles/firebolt-profile.md:
```diff
@@ -0,0 +1,86 @@
+**Maintained by:** Community  
+**Author:** Anders Swanson and Eric Ford
+**Source:** [Github](https://github.com/firebolt-db/dbt-firebolt)  
+**dbt Cloud:** Not Supported  
```
When you launch, we would be happy to provide a dbt slack channel if you would like. It would be called #db-firebolt

dataders replied:
or make us one now, so we can include it in this PR? #lazy

kevinmarr replied:
That would be great!

amychen1776 replied:
haha I can do that with the promise that you will maintain it :)

amychen1776 replied:
Here you go, here's the link 

https://getdbt.slack.com/archives/C02PYT5CXN0

The language is `dbt Slack channel: [Slack](https://getdbt.slack.com/archives/C02PYT5CXN0)`

ericf-firebolt replied:
10x!

ericf-firebolt committed:
Dimension table code has now been removed.

amychen1776 commented on website/sidebars.js:
```diff
@@ -515,7 +516,8 @@ module.exports = {
         "reference/warehouse-profiles/materialize-profile",
-        "reference/warehouse-profiles/rockset-profile"
+        "reference/warehouse-profiles/rockset-profile",
+        "reference/warehouse-profiles/firebolt-profile",
```
There's an extra comma here

ericf-firebolt committed:
Added a missing </Tabs> tag.

ericf-firebolt committed:
Last set of checks passed. Adding back end of file to see if there are additional issues.

ericf-firebolt committed:
Apply suggestions from code review

Changed final text still still referencing community support.

Co-authored-by: Anders <swanson.anders@gmail.com>

ericf-firebolt committed:
Removed supporting-concurrent-development information.

kevinmarr commented on website/docs/reference/resource-configs/firebolt-configs.md:
````diff
@@ -0,0 +1,339 @@
+  +quote_columns: false  #or `true` if you have csv column headers with spaces
+```
+
+#### Supporting Concurrent Development
````
Let's move this to the profiles doc

ericf-firebolt committed:
Moved quote_columns out of firebolt-configs and into firebolt-profile.

ericf-firebolt committed:
Finally got all the stuff in the right places in the profile and config files.

dataders commented on website/docs/reference/resource-configs/firebolt-configs.md:
````diff
@@ -0,0 +1,299 @@
+
+## Using dbt-firebolt
+
+### `quote_columns`
+
+To prevent a warning, you should add a configuration as below to your `dbt_project.yml`. For more info, see the [relevant dbt docs page](https://docs.getdbt.com/reference/resource-configs/quote_columns).
+
+
+<File name>
+```yml
+seeds:
+  +quote_columns: false  #or `true` if you have csv column headers with spaces
+```
+</File>
````
do we still need this section on `quote_columns`? my vote is to strike it.

something looks weird w/ it right now anyway

cc: @kevinmarr @amychen1776 @ericf-firebolt 

<img width="685" alt="image" src="https://user-images.githubusercontent.com/8158673/145123522-c9af34e8-25e0-445c-9d3d-882146630f7c.png">


```suggestion
```

amychen1776 replied:
Since this in the profiles config page, I don't think so.

ericf-firebolt replied:
The thing that was weird is that it said "yml" after the triple ticks and not "yaml".

dataders commented on website/docs/reference/resource-configs/firebolt-configs.md:
````diff
@@ -0,0 +1,299 @@
+
+3. Pull in the `packages.yml` dependencies by calling `dbt deps`.
+
+#### Using External Tables
+
+To use external tables, you must define a table as `external` in your `dbt_project.yml` file. Every external table must contain the fields `url`, `type`, and `object_pattern`. Note that the Firebolt external table specification requires fewer fields than what is specified in the dbt documentation.
+
+In addition to specifying the columns, an external table may specify partitions. Partitions are not columns and they cannot have the same name as columns. To avoid yaml parsing errors, remember to encase string literals (such as the `url` and `object_pattern` values) in single quotation marks.
+
+#### dbt_project.yml Syntax For an External Table
+
+```yaml
+sources:
+  - name: firebolt_external
+    schema: "{{ target.schema }}"
+    loader: S3
+
+    tables:
+      - name: <table-name>
+        external:
+          url: 's3://<bucket_name>/'
+          object_pattern: '<regex>'
+          type: '<type>'
+          credentials:
+            internal_role_arn: arn:aws:iam::id:<role>/<bucket-name>
+            external_role_id: <external-id>
+          object_pattern: '<regex>'
+          compression: '<compression-type>'
+          partitions:
+            - name: <partition-name>
+              data_type: <partition-type>
+              regex: '<partition-definition-regex>'
+          columns:
+            - name: <column-name>
+              data_type: <type>
+```
````
we've already gutted most of the "using dbt-external tables" section, I vote we get rid of the rest of it?
```suggestion
```

amychen1776 commented on website/docs/reference/warehouse-profiles/firebolt-profile.md:
```diff
@@ -0,0 +1,99 @@
+
+Some core functionality may be limited. If you're interested in contributing, check out the source code for each repository listed below.
+
+:::
```
```suggestion

```

amychen1776 commented on website/docs/reference/warehouse-profiles/firebolt-profile.md:
```diff
@@ -0,0 +1,99 @@
+---
+
+
+Some core functionality may be limited. If you're interested in contributing, check out the source code for each repository listed below.
```
```suggestion
Some core functionality may be limited. If you're interested in contributing, check out the source code for the repository listed below.
```

amychen1776 commented on website/docs/reference/warehouse-profiles/firebolt-profile.md:
```diff
@@ -0,0 +1,99 @@
+## Overview of dbt-firebolt
+
+**Maintained by:** Firebolt  
+**Author:** Anders Swanson and Eric Ford
```
```suggestion
**Author:** Anders Swanson and Eric Ford     
```

amychen1776 replied:
Need whitespace to help with the spacing

amychen1776 commented on website/docs/reference/warehouse-profiles/firebolt-profile.md:
```diff
@@ -0,0 +1,99 @@
+
+### Connecting to Firebolt
+
+To connect to Firebolt from dbt, you'll need to add a new [profile](https://docs.getdbt.com/dbt-cli/configure-your-profile) to your `profiles.yml` file. A Firebolt profile conforms to the following syntax:
```
```suggestion
To connect to Firebolt from dbt, you'll need to add a [profile](https://docs.getdbt.com/dbt-cli/configure-your-profile) to your `profiles.yml` file. A Firebolt profile conforms to the following syntax:
```

amychen1776 commented on website/docs/reference/warehouse-profiles/firebolt-profile.md:
```diff
@@ -0,0 +1,99 @@
+- If you're not using the default engine for the database, you must specify an engine name.
+- If there is more than one account associated with your credentials, you must specify an account.
+
+To connect to Firebolt from dbt, you'll need to add a new [profile](https://docs.getdbt.com/dbt-cli/configure-your-profile) to your `profiles.yml` file. A Firebolt profile conforms to the following syntax:
```
```suggestion

```

amychen1776 replied:
this is duplicated

amychen1776 commented on website/docs/reference/warehouse-profiles/firebolt-profile.md:
```diff
@@ -0,0 +1,99 @@
+
+For an example of how this works, let’s say Shahar and Eric are both working on the same project.
+
+In her `.dbt/profiles.yml`, Sharar sets `schema=sh`, whereas Eric sets `schema=er` in his. When each runs the `customers` model, the models will land in the database as tables named `sh_customers` and `er_customers`, respectively. When running dbt in production, you would use yet another `profiles.yml` with nothing set in the `schema` field (or any string of your choice).
```
By another profiles.yml, do you mean actually another target in the profile for rockset?

kevinmarr replied:
I assume that when the model is run in production, it would run on some server in the cloud somewhere that would have its own profiles.yml file, separate from the individual developers.

kevinmarr replied:
@amychen1776 Let me know if this is incorrect

amychen1776 commented on website/docs/reference/resource-configs/firebolt-configs.md:
```diff
@@ -0,0 +1,299 @@
+
+### Model Configuration for Fact Tables
+
+A dbt model can be created as a Firebolt fact table and configured using the following syntax:
```
```suggestion
A dbt model be materialized as a Firebolt fact table and configured using the following syntax:
```

amychen1776 commented on website/docs/reference/resource-configs/firebolt-configs.md:
````diff
@@ -0,0 +1,299 @@
+
+```yaml
+models:
+  [<resource-path>]:
````
Should these brackets be here (for resource-path)?

kevinmarr replied:
No - not sure how those got in because it is not displayed that way in our [readme](https://github.com/firebolt-db/dbt-firebolt).

amychen1776 commented on website/docs/reference/resource-configs/firebolt-configs.md:
```diff
@@ -0,0 +1,299 @@
+</TabItem>
+
+<TabItem value="property-yaml">
+<File name='models/properties.yml'>
```
Wait, this looks odd. @swanderz Users should only be applying model configurations on the model config block and the dbt_project.yml file. This is not a blocker for me but feels odd to recommend to a user from a workflow perspective (why have a model file with configs when the logic is easier for the user in the model file or project.yml for massive application)

dataders replied:
@kevinmarr @ericf-firebolt just chatted w/ @amychen1776 on this, and the dbt perspective is that while yes, you can set model configs in property files, it really isn't recommended. Her advice is to strike the property file examples for both dimension and fact tables

kevinmarr replied:
Great, let's axe 'em.

ericf-firebolt committed:
Removed text saying you could have an empty schema name.

ericf-firebolt committed:
Committing Amy's typos and formatting changes.

Co-authored-by: Amy Chen <46451573+amychen1776@users.noreply.github.com>

ericf-firebolt committed:
Merge branch 'current' of github.com:firebolt-db/docs.getdbt.com into current

ericf-firebolt committed:
Removed property files spec from firebolt-configs.md.

ericf-firebolt committed:
Added Slack channel to firebolt-profile.md.

ericf-firebolt committed:
Small typo.

ericf-firebolt committed:
Removed extraneous brackets and "using external tables."

Co-authored-by: Anders <swanson.anders@gmail.com>
Co-authored-by: Amy Chen <46451573+amychen1776@users.noreply.github.com>

ericf-firebolt committed:
Removed extraneous brackets.

ericf-firebolt committed:
Merge branch 'current' of github.com:firebolt-db/docs.getdbt.com into current

ericf-firebolt committed:
Removed quote_columns from firebolt_config.md.

Co-authored-by: Anders <swanson.anders@gmail.com>

kevinmarr commented on website/docs/reference/resource-configs/firebolt-configs.md:
```diff
@@ -0,0 +1,299 @@
+id: "firebolt-configs"
+---
+
+## Using dbt-firebolt
```
@ericf-firebolt This is an example of a header that is no longer relevant due to the change in context between the dbt docs vs our readme - please remove

kevinmarr commented on website/docs/reference/resource-configs/firebolt-configs.md:
````diff
@@ -0,0 +1,299 @@
+
+
+<File name>
+```yml
````
@ericf-firebolt This is not rendering properly on the rich diff in GH, which doesn't necessarily mean it won't render properly on dbt docs, but I just wanted to flag that. Might need some newlines.

kevinmarr commented on website/docs/reference/resource-configs/firebolt-configs.md:
````diff
@@ -0,0 +1,299 @@
+) }}
+```
+
+### Model Configuration for Dimension Tables
````
I realized that we left out `primary_index` in the dimension tables section. @ericf-firebolt Could you add it in? It will need to be added to the code examples as well as the table.
